code/modules/projectiles/gun.dm

```diff
@@ -977,16 +977,18 @@
 		playsound(src, fire_sound, 100, TRUE)
 
 /mob/living/proc/trip_with_gun()
+	var/chance_to_fire = GUN_NO_SAFETY_MALFUNCTION_CHANCE_HIGH
+	var/mob/living/carbon/human/human_holder
+	if(ishuman(src))
+		holder = src
```

Comment:
2 things:
1. this should be "human_holder = src", not "holder = src"
2. assigning a reference to a new variable that is typed is, typically, free. that can still sit inside the loop, not that it makes a huge difference here since you're using the null value to represent "they weren't a human"

Reply:
whoops, my bad. Thanks for the catch